src/z_zone.c

```diff
@@ -46,10 +49,10 @@ typedef struct memblock {
   struct memblock *next,*prev;
   size_t size;
   void **user;
-  unsigned char tag;
+  int tag;
```

Comment:
If we care about alignment, is it 64-bit or 32-bit version? With 64-bit build currently we have:
```
typedef struct memblock {
  unsigned id;                 // 4 bytes
  struct memblock *next,*prev; // 2 * 8 bytes
  size_t size;                 // 8 bytes
  void **user;                 // 8 bytes
  int tag;                     // 4 bytes
} memblock_t;
```
I always forget the exact data alignment/padding rules, usually I just reorder the fields by size. For the 64-bit version, there are a lot of padding the old Doom structs.

Reply:
Chunk size is currently 64 byte on a 64-bit system. Maybe that's a bit too much.

Reply:
Should we care about alignment in structures? If yes, then we need to reorder the `struct memblock`, currently there is padding:
```
typedef struct memblock {
  unsigned id;                 // 4 bytes
  // pad 4 bytes
  struct memblock *next,*prev; // 2 * 8 bytes
  size_t size;                 // 8 bytes
  void **user;                 // 8 bytes
  int tag;                     // 4 bytes
} memblock_t;
```
This is my favorite article on subject: http://www.catb.org/esr/structure-packing/#_structure_alignment_and_padding